query/command.go

```
@@ -112,6 +113,21 @@ func (cmd *SelectCommand) Execute(context ExecutionContext) (interface{}, error)
 	if err != nil {
 		return nil, err
 	}
+	slotLimit := context.SlotLimit
+	defaultLimit := 1000
+	if slotLimit == 0 {
+		slotLimit = defaultLimit // the default limit
+	}
+	if timerange.Slots() > slotLimit {
+		var limitMessage string
+		if context.SlotLimit == 0 {
+			limitMessage = "the default limit %d (no limit has been configured)"
+		} else {
+			limitMessage = "the configured limit %d"
+		}
+		limitMessage = fmt.Sprintf(limitMessage, slotLimit)
+		return nil, fmt.Errorf("Requested number of data points (%d) exceeds %s", timerange.Slots(), limitMessage)
```

we should throw an error similar to the error we get from the execution error.